core/io/resource.cpp

```diff
@@ -539,6 +539,12 @@ Ref<Resource> ResourceCache::get_ref(const String &p_path) {
 	return ref;
 }
 
+void ResourceCache::erase_ref(const String &p_path) {
+	ResourceCache::lock.lock();
+	ResourceCache::resources.erase(p_path);
+	ResourceCache::lock.unlock();
```

Comment:
We're already in `ResourceCache`:
```suggestion
	lock.lock();
	resources.erase(p_path);
	lock.unlock();
```